config/routes.rb

```diff
@@ -1,5 +1,8 @@
 Rails.application.routes.draw do
- root 'items#index'
- get  "/items/new" =>  "items#new"
- post "/items"     =>  "items#create"
+  root 'items#index'
+  get  "/items/new"                         =>  "items#new"
+  post "/items"                             =>  "items#create"
+  get  "/items/:id/purchases/new"           =>  "purchases#new", as: 'new_purchase'
```

Comment:
Rename the :id param here to :item_id to be consistent across the purchases controller.